### PR TITLE
Get registry from v2prov even without a secret

### PR DESCRIPTION
**Pull request by felipe-colussi**
## Problem

This was happening wile creating a RKE2/K3S downstream cluster with a different public registry that the upstream cluster. 

Both webhook and the Helm/Shell are created from the Rancher Agent. 

The code that created the rancher agent template was only getting the v2prov registry if Auth was set.

With that for public registries it was  getting the upstream default registry instead of the downstream one. 

 
## Solution

Change the logic to get the RegistryUrl so it checks the RKE2 even if it doesn't have auth. 

 
### Manual Testing

Create RKE1 and RKE2 clusters with custom and non custom registries. 

### Automated Testing

* Test types added/modified:
    * Unit: 
       Added some cases: 
       * Registry set for upstream and downstream cluster with secret. 
       * Registry set for upstream and downstream cluster without secret.
       * Registry set for upstream cluster and not for downstream cluster.  
       

## QA Testing Considerations
Creation of RKE clusters with distinct registries. 

Despite the issue being reported for clusters without secrets I had to change some code execution so please also test with private registry. 
 
### Regressions Considerations 
I Tried my best to keep the same behavior as the existing one.  I don't think we can have a regression with this one.

### CODE REVIEW

File: pkg/cluster/private_registry.go
Comment (HarrisonWAffel) [inline pkg/cluster/private_registry.go:133]: Just to add a bit more explanation, this condition will happen if the v2 prov registry exists, but is _unauthenticated_.

File: pkg/cluster/private_registry_test.go
Comment (HarrisonWAffel) [inline pkg/cluster/private_registry_test.go:170]: Could we rename this to `global system default registry and no cluster default registry`? It can get pretty confusing if we don't stick to the current terminology

**Review by HarrisonWAffel — APPROVED** [review]
LGTM! thanks for adding the additional comments and renaming the variables, makes it easier to understand. 

Please remember to squash all of the commits in this PR (and to wait for a second approval :) )

File: pkg/cluster/private_registry.go
Comment (jakefhyde) [inline pkg/cluster/private_registry.go:70]:
```suggestion
	// Declare here so we don't need to check if the rkeClusterRegistryOrGlobalSystemDefault exists while working with v2prov
```

File: pkg/cluster/private_registry_test.go
Comment (jakefhyde) [inline pkg/cluster/private_registry_test.go:148]:
```suggestion
			name:           "global system default registry and cluster default registry without secret",
```

File: pkg/cluster/private_registry_test.go
Comment (jakefhyde) [inline pkg/cluster/private_registry_test.go:114]:
```suggestion
			name:           "global system default registry and no cluster default registry with secret",
```

**Review by jakefhyde — CHANGES_REQUESTED** [review]
Good job adding comments and tests. The logic was hard to follow (not your fault, its just complex) but I was able to figure out whats going on because of that. Just a few spelling nits.

**Review by jakefhyde — APPROVED** [review]

**Review by HarrisonWAffel — APPROVED** [review]

**Pull request merged**
